packages/contracts/contracts/L1/L2OutputOracle.sol

```diff
@@ -219,15 +221,23 @@ contract L2OutputOracle is Initializable, Semver {
             );
         }
 
-        emit OutputSubmitted(_outputRoot, nextOutputIndex(), _l2BlockNumber, block.timestamp);
+        uint256 outputIndex = nextOutputIndex();
+        emit OutputSubmitted(_outputRoot, outputIndex, _l2BlockNumber, block.timestamp);
```

Comment:
isn't it better to emit the event at the end of this function?

Reply:
Existing code also emit in this position. However, as you said, emitting after a state change seems like a good idea.

Reply:
@chokobole @0xHansLee wdyt?

Reply:
I also agreed.

Reply:
agreed